api/environments/identities/models.py

```diff
@@ -107,9 +107,11 @@ def get_all_feature_states(self, traits: typing.List[Trait] = None):
                 current_flag = identity_flags[flag.feature_id]
                 if flag > current_flag:
                     identity_flags[flag.feature_id] = flag
-
-        if self.environment.project.hide_disabled_flags:
-            # filter out any flags that are disabled if configured on the project
+        if (
+            self.environment.hide_disabled_flags
```

Comment:
I think I'd like to see this logic as a method on the environment model so that we can unit test it with a parametrised test to fully describe the functionality? 

So the test would look something like: 

```python
@pytest.mark.parametrize(
  "environment_value, project_value, expected_result",
  (
    (True, True, True),
    (True, False, False),
    (False, True, False),
    (False, False, False),
  )
)
def test_get_hide_disabled_flags(environment_value, project_value, expected_result, organisation):
    project = Project.objects.create(name="test", organisation=organisation, hide_disabled_flags=project_value)
    environment = Environment.objects.create(name="test", project=project, hide_disabled_flags=environment_value)
    assert environment.get_hide_disabled_flags() is expected_result
```

Reply:
In writing this (which was the reason I actually wrote out the test), I actually think the logic might be slightly wrong here. I think the project value should be a 'default' that can be overridden by the environment. I think we talked about allowing null values for the environment level value to signify 'not set'. 

The reason we're adding this is primarily for a customer to be able to test enabling it in a single environment so they don't have to enable it at the project level for all of their environments. 

So, I think the test cases should actually be: 

```python
@pytest.mark.parametrize(
  "environment_value, project_value, expected_result",
  (
    (True, True, True),
    (True, False, True),
    (False, True, False),
    (False, False, False),
    (None, True, True),
    (None, False, False),
  )
)
```

Then the UI just has a drop down instead of a switcher that has the following options: 

 - use project value
 - hide disabled flags
 - show disabled flags

If you think that using an enum would be more appropriate based on this then I'm fine with that too - although I think my preference would be the null / boolean approach.